carts/views.py

```diff
@@ -0,0 +1,163 @@
+from itertools import product
+import re
+import json
+
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import F
+
+from carts.models import Cart
+from users.models import User
+from products.models import ProductOption
+
+def validate_cartdata(func) :
+    def wrapper(self, request) :
+        try :
+            data        = json.loads(request.body)
+            data        = data["message"]
+            
+            user_id             = data["user_id"]
+            product_option_id   = data["product_options_id"]
+            user            =  User.objects.get(id = user_id)
+            product_option  = ProductOption.objects.get(id = product_option_id)
+        
+        except json.JSONDecodeError :
+            return JsonResponse({"message" : "INVALID_BODY_REQEUST"}, status = 400)
+        except KeyError :
+            return JsonResponse({"message" : "KEY_ERROR"}, status = 400)
+        except User.DoesNotExist :
+            return JsonResponse({"message" : "INVALID_USER"}, status = 404)
+        except ProductOption.DoesNotExist :
+            return JsonResponse({"message" : "INVLALID_PRODUCT"}, status = 404)
+        return func(self, request)
+    return wrapper
+
+def validate_quantity(func) :
+    REX_INTEGER = r'^[+]{0,1}\d+$'
+    def wrapper(self , request,**kargs) :
+        try :
+            data        = json.loads(request.body)
+            data        = data["message"]
+            quantity    = str(data["product_count"])
+
+            if not re.fullmatch(REX_INTEGER,quantity) :
+                return JsonResponse({"message" : "INVALID_VALUE"}, status = 400)
+
+            check_integer = int(quantity)
+            if check_integer <= 0 or check_integer > 4294967295 : 
+                return JsonResponse({"message" : "INVALID_QUANTITY_RANGE"}, status = 400)
+        
+        except json.JSONDecodeError :
+            return JsonResponse({"message" : "INVALID_BODY_REQEUST"}, status = 400)
+        except KeyError :
+            return JsonResponse({"message" : "KEY_ERROR"}, status = 400)      
+        return func(self, request,**kargs)
+    return wrapper
+
+class GetCartListView(View) :
+    def get(self, request, user_id) :
+
+        user    = user_id
+        items   = Cart.objects.filter(user_id = user).all().select_related('product').all().select_related('product').all()
```

Comment:
select_related 함수를 왜 두번 사용한건가요?
```suggestion
        items   = Cart.objects.filter(user_id = user).all().select_related('product').all().select_related('product').all()
```

Reply:
1. Carts(장바구니)모델에 있는 product 어트리뷰트는 정확하게 product_option 테이블의 id를 외래키로 참조합니다.
2. product_Option 모델 product 어트리뷰트는 Product 모델 id에 접근하는 외래키 어트리뷰트 입니다
3. 결론적으로 Product모델 안에 있는 name(제품이름을 저장하는 어트리뷰트)을 가져와야 해서
4. 2번 정참조를 위해서  (Cart --['product']--> Product_Option--['product']'--> Product) 사용했습니다